world-explore/src/app/pipes/groupByLetter.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Country } from '../models/model';
+import { CountryCode } from '../shared/country-code.enum';  // Import the enum from the shared folder
+
+@Pipe({
+  name: 'groupByLetter'
+})
+export class GroupByLetterPipe implements PipeTransform {
+  transform(countries: Country[]): { [key: string]: Country[] } {
+    const groupedCountries: { [key: string]: Country[] } = {};
+
+    countries.forEach(country => {
```

Comment:
utilize array methods such as filter, map, and reduce  as below : 
1- Filtering:
Uses countries.filter(country => country.iso3 !== CountryCode.ISR) to exclude countries where iso3 is CountryCode.ISR without using an if statement explicitly.

2-Reducing into Groups:
Utilizes reduce to accumulate countries into groupedCountries based on the first letter of country.country.

3-Grouping Logic:
groupedCountries[firstLetter] = [...(groupedCountries[firstLetter] || []), country]; efficiently groups countries by their first letter using array spread syntax.